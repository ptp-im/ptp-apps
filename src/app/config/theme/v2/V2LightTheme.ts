import color from 'color';
import { MD2LightTheme } from 'react-native-paper';
import type { MD2Theme } from 'react-native-paper';

import { black, red800, white } from './colors';

export const V2LightTheme: MD2Theme = {
  ...MD2LightTheme,
  dark: false,
  roundness: 4,
  version: 2,
  isV3: false,
  colors: {
    ...MD2LightTheme.colors,
    // primary: '#6200ee',
    accent: '#03dac4',
    // background: '#f6f6f6',
    // background: grey50,
    surface: white,
    error: '#B00020',
    // text: black,
    // onSurface: '#000000',
    disabled: color(black).alpha(0.26).rgb().string(),
    placeholder: color(black).alpha(0.54).rgb().string(),
    backdrop: color(black).alpha(0.5).rgb().string(),
    // notification: pinkA400,

    msgUnreadBadge: red800,
    onSurface: 'rgb(30, 107, 192)',
    primary: 'rgb(30, 107, 192)',
    background: 'rgb(235, 235, 240)',
    white: 'rgb(255, 255, 255)',
    background1: 'rgb(235, 235, 240)',
    backgroundSelected: 'rgb(62, 125, 185)',
    backgroundTabs: 'rgb(255, 255, 255)',
    card: 'rgb(255, 255, 255)',
    text: 'rgb(28, 28, 30)',
    textPrimary: 'rgb(10, 132, 255)',
    textAlert: 'rgb(255, 59, 48)',
    border: 'rgb(229, 229, 229)',
    notification: 'rgb(255, 59, 48)',
  },
};

export const V2NavLightColors = {
  primary: V2LightTheme.colors.primary,
  background: V2LightTheme.colors.background,
  card: V2LightTheme.colors.card,
  text: V2LightTheme.colors.text,
  border: V2LightTheme.colors.border,
  notification: V2LightTheme.colors.notification,
};
