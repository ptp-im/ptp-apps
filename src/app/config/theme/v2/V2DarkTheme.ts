import color from 'color';

import type { MD2Theme } from 'react-native-paper';
import { black, red800, white } from './colors';
import { V2LightTheme } from './V2LightTheme';

export const V2DarkTheme: MD2Theme = {
  ...V2LightTheme,
  dark: true,
  mode: 'adaptive',
  version: 2,
  isV3: false,
  colors: {
    ...V2LightTheme.colors,
    // primary: '#BB86FC',
    accent: '#03dac6',
    // background: '#121212',
    surface: '#121212',
    error: '#CF6679',
    onSurface: '#FFFFFF',
    // text: white,
    disabled: color(white).alpha(0.38).rgb().string(),
    placeholder: color(white).alpha(0.54).rgb().string(),
    backdrop: color(black).alpha(0.5).rgb().string(),
    // notification: pinkA100,

    msgUnreadBadge: red800,
    primary: 'rgb(10, 132, 255)',
    background: 'rgb(1, 1, 1)',
    card: 'rgb(18, 18, 18)',
    text: 'rgb(229, 229, 231)',
    textPrimary: 'rgb(10, 132, 255)',
    border: 'rgb(39, 39, 41)',
    notification: 'rgb(255, 69, 58)',
  },
};

export const V2NavDarkColors = {
  primary: V2LightTheme.colors.primary,
  background: V2LightTheme.colors.background,
  card: V2LightTheme.colors.card,
  text: V2LightTheme.colors.text,
  border: V2LightTheme.colors.border,
  notification: V2LightTheme.colors.notification,
};
