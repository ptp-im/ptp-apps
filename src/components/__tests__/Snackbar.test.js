import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import renderer from 'react-test-renderer';

import { red200, white } from '../../styles/themes/v2/colors';
import Snackbar from '../Snackbar.tsx';

const styles = StyleSheet.create({
  snackContent: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  iconView: {
    backgroundColor: red200,
    padding: 15,
  },
  text: { color: white, marginLeft: 10, flexWrap: 'wrap', flexShrink: 1 },
});

// Make sure any animation finishes before checking the snapshot results
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.Animated.timing = (value, config) => ({
    start: (callback) => {
      value.setValue(config.toValue);
      callback && callback({ finished: true });
    },
  });

  return RN;
});

it('renders snackbar with content', () => {
  const tree = renderer
    .create(
      <Snackbar visible onDismiss={jest.fn()}>
        Snackbar content
      </Snackbar>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders not visible snackbar with content wrapper but no actual content', () => {
  const tree = renderer
    .create(
      <Snackbar visible={false} onDismiss={jest.fn()}>
        Snackbar content
      </Snackbar>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders snackbar with Text as a child', () => {
  const tree = renderer
    .create(
      <Snackbar visible onDismiss={jest.fn()}>
        <Text>Snackbar content</Text>
      </Snackbar>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders snackbar with action button', () => {
  const tree = renderer
    .create(
      <Snackbar
        visible
        onDismiss={() => {}}
        action={{ label: 'Undo', onPress: jest.fn() }}
      >
        Snackbar content
      </Snackbar>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders snackbar with View & Text as a child', () => {
  const tree = renderer
    .create(
      <Snackbar visible onDismiss={jest.fn()}>
        <View style={styles.snackContent}>
          <View style={styles.iconView} />
          <Text style={styles.text}>
            Error Message which is veryyyyyyyyyyyy longggggggg Error Message
            which is veryyyyyyyyyyyy longggggggg
          </Text>
        </View>
      </Snackbar>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
