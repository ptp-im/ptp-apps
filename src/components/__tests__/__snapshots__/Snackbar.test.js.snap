// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders not visible snackbar with content wrapper but no actual content 1`] = `null`;

exports[`renders snackbar with Text as a child 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 6,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        }
      }
    >
      <View
        accessibilityLiveRegion="polite"
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(28, 27, 31, 1)",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "margin": 8,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "marginLeft": 16,
                "marginVertical": 14,
              },
              Object {
                "color": "rgba(255, 251, 254, 1)",
                "marginRight": 16,
              },
            ]
          }
        >
          <View>
            <Text>
              Snackbar content
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`renders snackbar with View & Text as a child 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 6,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        }
      }
    >
      <View
        accessibilityLiveRegion="polite"
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(28, 27, 31, 1)",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "margin": 8,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "marginLeft": 16,
                "marginVertical": 14,
              },
              Object {
                "color": "rgba(255, 251, 254, 1)",
                "marginRight": 16,
              },
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#ef9a9a",
                    "padding": 15,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "flexShrink": 1,
                    "flexWrap": "wrap",
                    "marginLeft": 10,
                  }
                }
              >
                Error Message which is veryyyyyyyyyyyy longggggggg Error Message which is veryyyyyyyyyyyy longggggggg
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`renders snackbar with action button 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 6,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        }
      }
    >
      <View
        accessibilityLiveRegion="polite"
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(28, 27, 31, 1)",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "margin": 8,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "flex": 1,
                  "marginLeft": 16,
                  "marginVertical": 14,
                },
                Object {
                  "color": "rgba(255, 251, 254, 1)",
                  "marginRight": 0,
                },
              ],
            ]
          }
        >
          Snackbar content
        </Text>
        <View
          collapsable={false}
          style={
            Object {
              "alignSelf": undefined,
              "bottom": undefined,
              "left": undefined,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "top": undefined,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "marginHorizontal": 8,
                  "marginVertical": 6,
                  "minWidth": "auto",
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    false,
                    Object {
                      "borderRadius": 20,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          Object {
                            "marginHorizontal": 12,
                          },
                          Object {
                            "marginHorizontal": 8,
                          },
                          false,
                          Object {
                            "color": "rgba(208, 188, 255, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    Undo
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`renders snackbar with content 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 6,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        }
      }
    >
      <View
        accessibilityLiveRegion="polite"
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(28, 27, 31, 1)",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "margin": 8,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "flex": 1,
                  "marginLeft": 16,
                  "marginVertical": 14,
                },
                Object {
                  "color": "rgba(255, 251, 254, 1)",
                  "marginRight": 16,
                },
              ],
            ]
          }
        >
          Snackbar content
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
